adapters/quick/src/main/scala/caliban/QuickRequestHandler.scala

```diff
@@ -42,14 +42,25 @@ final private class QuickRequestHandler[R](
   def configureWebSocket[R1](config: quick.WebSocketConfig[R1]): QuickRequestHandler[R & R1] =
     new QuickRequestHandler[R & R1](interpreter, config)
 
-  def handleHttpRequest(request: Request)(implicit trace: Trace): URIO[R, Response] = ZIO.suspendSucceed {
-    transformHttpRequest(request)
-      .flatMap(executeRequest(request.method, _))
-      .foldZIO(
-        Exit.succeed,
-        resp => Exit.succeed(transformResponse(request, resp))
+  def handleHttpRequest(forwardUploadRequests: Boolean = false)(request: Request)(implicit
+    trace: Trace
+  ): URIO[R, Response] =
+    if (
+      forwardUploadRequests && request.headers.exists(
+        _.renderedValue.contains(MediaType.multipart.`form-data`.fullType)
```

Comment:
```suggestion
      forwardUploadRequests &&
      request.body.mediaType.exists(MediaType.multipart.`form-data`.matches(_, ignoreParameters = true))
```

I think this should be a lot more performant as it doesn't require iterating through all the headers. AFAIK the content type should automatically be attached to the body by `zio-http`